akd/agents/storm/storm.py

```diff
@@ -1,73 +1,116 @@
-from langgraph.pregel import RetryPolicy
-from pydantic import Field, computed_field
+from langgraph.types import RetryPolicy
+from loguru import logger
+from pydantic import AliasChoices, AnyUrl, Field
 
 from akd._base import InputSchema, OutputSchema
 from akd.agents._base import BaseAgent, BaseAgentConfig
-from akd.configs.storm_config import STORM_SETTINGS, StormSettings
+from akd.agents.search.aspect_search import AspectSearchConfig
 
-from .config import initialise_storm_config
 from .nodes import (
     conduct_interviews,
-    hitl_editors,
     index_references,
     initialize_research,
     refine_outline,
     write_article,
     write_sections,
 )
-from .state import ResearchState
-from .utils.outline_utils import Perspectives
+from .structures import Outline, Perspectives, ResearchState
 
 
 class StormInputSchema(InputSchema):
     """Input schema for storm agent"""
 
     config: dict | None = Field(
-        None,
+        default={"configurable": {"thread_id": "default-thread"}},
         description="The configuration dict to track the state of the storm agent.",
     )
-
-    query: str = Field(
+    topic: str = Field(
         ...,
+        validation_alias=AliasChoices("topic", "query"),
         description="The topic to create the article for.",
     )
-
-    @computed_field
-    @property
-    def topic(self) -> str:
-        return self.query
+    outline_sketch: str = Field(
+        default=None,
+        description="Rough sketch of the article's outline.",
+    )
 
 
 class StormOutputSchema(OutputSchema):
     """Output schema for storm agent"""
 
     article: str = Field(
         ...,
-        description="The article generated by storm.",
+        description="The article generated by storm",
     )
     perspectives: Perspectives = Field(
         ...,
         description="List of editors working on the article",
     )
-    interview_results: list = Field(
+    references: dict = Field(
+        ...,
+        description="References collected by aspect search",
+    )
+    search_results: list = Field(
         ...,
-        description="Interview results between editors",
+        description="Search results collected by aspect search",
+    )
+    outline: Outline = Field(
+        ...,
+        description="Outline of the article",
+    )
+    sections: list = Field(
+        ...,
+        description="List of sections of the article",
     )
 
 
 class StormAgentConfig(BaseAgentConfig):
     """Configuration for Storm Agent"""
 
-    hitl: bool = Field(
-        False,
-        description="Whether to set the agent in HITL mode.",
+    long_context_llm: str = Field(
+        default="gpt-4o-mini",
+        description="Model used for longer context based operations",
     )
-
-    storm_settings: StormSettings | None = Field(
-        default_factory=lambda: STORM_SETTINGS,
-        description="The configuration for the storm agent.",
+    aspect_search_config: Optional[object] = Field(
```

Comment:
let's use python 3.12+ new style to avoid importing Optional and other typing elements. eg: `AspectSearchConfig | None`